archinstall/lib/user_interaction.py

```diff
@@ -176,6 +192,25 @@ def ask_to_configure_network():
 			if not len(gateway := input('Enter your gateway (router) IP address or leave blank for none: ').strip()):
 				gateway = None
 
+			# Assuming that gateway (router) IP address doesn't contain subnet,
+			# we can implement this check for it
+			# Implemented new check for correct IP input
+
+			#gateway = input('Enter your gateway (router) IP address or leave blank for none: ').strip()
+			#while 1:
+			#	try:
+			#		if len(gateway) == 0:
+			#			gateway = None
+			# 		else:
+			# 			ipaddress.ip_address(gateway)
+			# 		break
+			# 	except ValueError:
+			# 		log(
+			# 			"You need to enter a valid gateway (router) IP address.",
+			# 			level=LOG_LEVELS.Warning,
+			# 			fg='red'
+			# 		)
+
```

Comment:
Should I uncomment this change because I'm still unsure if we are checking for regular IP and not for IP/subnet

Reply:
The gateway needs to be asked if static IP was chosen.
Otherwise there will be no internet for the poor user :)